client/src/BibleEngineClient.ts

```diff
@@ -22,6 +23,43 @@ export class BibleEngineClient {
         if (bibleEngineOptions) this.localBibleEngine = new BibleEngine(bibleEngineOptions);
     }
 
+    static chunk(arr: any, chunkSize = 1, cache: any[] = []) {
+        const tmp = [...arr];
+        if (chunkSize <= 0) return cache;
+        while (tmp.length) cache.push(tmp.splice(0, chunkSize));
+        return cache;
+    }
+
+    static async getBookIndexFile(versionUid: string, baseS3Url: string) {
+        const fileIndexUrl = `${baseS3Url}/${versionUid}/index.json`;
+        const response = await fetch(fileIndexUrl);
+        const fileIndex = await response.json();
+        return fileIndex;
+    }
+
+    async downloadVersion(versionUid: string, baseS3Url: string) {
+        console.time('downloadVersion');
+        const books = await BibleEngineClient.getBookIndexFile(versionUid, baseS3Url);
+        const params = { versionUid };
+        const { result } = await this.remoteApi!.getVersion(params);
+        const localVersion = await this.localBibleEngine.addVersion(result);
+        const BATCH_SIZE = 10;
+        const chunks = BibleEngineClient.chunk(books, BATCH_SIZE);
+        for (const chunk of chunks) {
+            const responses = await Promise.all(
```

Comment:
since you this code is about optimising overall download/import time: i think in most cases the longest wait is caused by importing the data into the schema (given a fast connection and being on a mobile device). so i would expect parallelising download and import to result in fastest overall import times (you basically eliminate download time - and even on a slower connection it should be faster than waiting for all BATCH_SIZE requests to have finished before starting to import, and then wait for for all imports before downloading the next batch):

```
let bookNum = 1;
const bookData: IBibleBook[] = [];

// loop logic

const waitFor: Promise[] = [];
if(bookNum <= lastBookNum) waitFor.push(downloadBook(bookNum));
if(bookData[bookNum-1]) waitFor.push(importBook(bookData[bookNum-1]));
await Promise.all(waitFor);
```